In [1]:
# Initial imports
import pandas as pd
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
import numpy as np

## Loading and Preprocessing WHO Data

In [2]:
# Loading data
who_data_df = pd.read_csv("../Datasets_ML/WHO_country_ML_wNaN.csv")
who_data_df

DIM_GEO_NAME DIM_GEO_CODE  region          sub-region  DIM_TIME_YEAR  \
0    Afghanistan          AFG    Asia       Southern Asia           2016   
1    Afghanistan          AFG    Asia       Southern Asia           2018   
2    Afghanistan          AFG    Asia       Southern Asia           2019   
3    Afghanistan          AFG    Asia       Southern Asia           2021   
4    Afghanistan          AFG    Asia       Southern Asia           2022   
..           ...          ...     ...                 ...            ...   
922     Zimbabwe          ZWE  Africa  Sub-Saharan Africa           2019   
923     Zimbabwe          ZWE  Africa  Sub-Saharan Africa           2020   
924     Zimbabwe          ZWE  Africa  Sub-Saharan Africa           2021   
925     Zimbabwe          ZWE  Africa  Sub-Saharan Africa           2022   
926     Zimbabwe          ZWE  Africa  Sub-Saharan Africa           2023   

     AMR_INFECT_ECOLI  AMR_INFECT_MRSA  FINPROTECTION_CATA_TOT_10_POP  \
0                 NaN              NaN                            NaN   
1                 NaN              NaN                            NaN   
2                 NaN              NaN                            NaN   
3                 NaN              NaN                            NaN   
4                 NaN              NaN                            NaN   
..                ...              ...                            ...   
922               NaN              NaN                            NaN   
923               NaN              NaN                            NaN   
924               NaN              NaN                            NaN   
925               NaN              NaN                            NaN   
926               NaN              NaN                            NaN   

     FINPROTECTION_CATA_TOT_25_POP  GHED_GGHE-DGGE_SHA2011  ...  \
0                              NaN                     NaN  ...   
1                              NaN                     NaN  ...   
2                              NaN                     NaN  ...   
3                              NaN                    4.05  ...   
4                              NaN                     NaN  ...   
..                             ...                     ...  ...   
922                            NaN                     NaN  ...   
923                            NaN                     NaN  ...   
924                            NaN                    5.21  ...   
925                            NaN                     NaN  ...   
926                            NaN                     NaN  ...   

     UHC_INDEX_REPORTED  VACCINEPREVENTABLE_WILDPOLIO  VIOLENCE_HOMICIDERATE  \
0                   NaN                           NaN                    NaN   
1                   NaN                           NaN                    NaN   
2                   NaN                           NaN                    NaN   
3             40.884609                           NaN               8.378230   
4                   NaN                           NaN                    NaN   
..                  ...                           ...                    ...   
922                 NaN                           NaN                    NaN   
923                 NaN                           NaN                    NaN   
924           55.042839                           NaN              12.219641   
925                 NaN                           NaN                    NaN   
926                 NaN                           0.0                    NaN   

     WHOSIS_000003  WHS4_100  WSH_DOMESTIC_WASTE_SAFELY_TREATED  \
0              NaN       NaN                                NaN   
1              NaN       NaN                                NaN   
2              NaN       NaN                                NaN   
3              NaN       NaN                                NaN   
4        35.489811      69.0                                NaN   
..             ...       ...                  

# 1) Analysis without Geographical features

In [3]:
# Define features set
X = who_data_df.drop(columns=["DIM_GEO_NAME", "DIM_GEO_CODE", "region", "sub-region",
                              "DIM_TIME_YEAR", "Healthy life expectancy at birth"])
X.head()

AMR_INFECT_ECOLI  AMR_INFECT_MRSA  FINPROTECTION_CATA_TOT_10_POP  \
0               NaN              NaN                            NaN   
1               NaN              NaN                            NaN   
2               NaN              NaN                            NaN   
3               NaN              NaN                            NaN   
4               NaN              NaN                            NaN   

   FINPROTECTION_CATA_TOT_25_POP  GHED_GGHE-DGGE_SHA2011  GLASSAMC_AWARE  \
0                            NaN                     NaN             NaN   
1                            NaN                     NaN             NaN   
2                            NaN                     NaN             NaN   
3                            NaN                    4.05             NaN   
4                            NaN                     NaN             NaN   

   HWF_0001  HWF_0006  HWF_0010  HWF_0014  ...  \
0       NaN       NaN       NaN       NaN  ...   
1       NaN      4.52       NaN     0.292  ...   
2       NaN       NaN     0.714       NaN  ...   
3       NaN       NaN       NaN       NaN  ...   
4       NaN       NaN       NaN       NaN  ...   

   SUD_TREATMENTSERVICES_COVERAGE  UHC_INDEX_REPORTED  \
0                             NaN                 NaN   
1                             NaN                 NaN   
2                             NaN                 NaN   
3                             NaN           40.884609   
4                             NaN                 NaN   

   VACCINEPREVENTABLE_WILDPOLIO  VIOLENCE_HOMICIDERATE  WHOSIS_000003  \
0                           NaN                    NaN            NaN   
1                           NaN                    NaN            NaN   
2                           NaN                    NaN            NaN   
3                           NaN                8.37823            NaN   
4                           NaN                    NaN      35.489811   

   WHS4_100  WSH_DOMESTIC_WASTE_SAFELY_TREATED  WSH_HYGIENE_BASIC  \
0       NaN                                NaN                NaN   
1       NaN                                NaN                NaN   
2       NaN                                NaN                NaN   
3       NaN                                NaN                NaN   
4      69.0                                NaN          48.214699   

   WSH_SANITATION_SAFELY_MANAGED  WSH_WATER_SAFELY_MANAGED  
0                            NaN                       NaN  
1                            NaN                       NaN  
2                            NaN                       NaN  
3                            NaN                       NaN  
4                            NaN                   30.0341  

[5 rows x 56 columns]

In [4]:
# Define target vector
y = who_data_df["Healthy life expectancy at birth"]
y.head()

0    52.924999
1    53.575001
2    53.900002
3    54.549999
4    54.875000
Name: Healthy life expectancy at birth, dtype: float64

In [5]:
# Splitting into Train and Test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=78)

## Fitting the Random Forest Regressor

In [6]:
# Create the random forest regressor instance
rf_regressor = RandomForestRegressor(n_estimators=500, random_state=78)

In [7]:
# Fit the model
rf_regressor.fit(X_train, y_train)

RandomForestRegressor(n_estimators=500, random_state=78)

## Making Predictions Using the Random Forest Regressor Model

In [8]:
# Making predictions using the testing data
predictions = rf_regressor.predict(X_test)

## Model Evaluation

In [9]:
# Regression evaluation metrics using scikit-learn

# Mean Absolute Error (MAE):
mae = mean_absolute_error(y_test, predictions)
print(f'Mean Absolute Error (MAE): {mae}')

# Mean Squared Error (MSE):
mse = mean_squared_error(y_test, predictions)
print(f'Mean Squared Error (MSE): {mse}')

# Root Mean Squared Error (RMSE):
rmse = np.sqrt(mean_squared_error(y_test, predictions))
print(f'Root Mean Squared Error (RMSE): {rmse}')

# R-squared (R²):
r2 = r2_score(y_test, predictions)
print(f'R-squared (R²): {r2}')

Mean Absolute Error (MAE): 2.8177894075860634
Mean Squared Error (MSE): 14.681086360143938
Root Mean Squared Error (RMSE): 3.8315905783556703
R-squared (R²): 0.5588363436550626


## Mean Absolute Error (MAE):

Value: 2.82
Interpretation: On average, the model's predictions are off by about 2.82 years from the actual healthy life expectancy. This metric provides a straightforward interpretation of the average error in the same units as the target variable.

## Mean Squared Error (MSE):

Value: 14.68
Interpretation: This metric squares the errors before averaging, which means larger errors have a disproportionately larger impact. It indicates the average of the squared differences between predicted and actual values, suggesting the model's overall error magnitude.

## Root Mean Squared Error (RMSE):

Value: 3.83
Interpretation: RMSE is the square root of MSE and is also in the same units as the target variable. It indicates that the model's predictions deviate from actual values by about 3.83 years on average. RMSE is sensitive to outliers, so it can be a good measure of model performance when large errors are particularly undesirable.

## R-squared (R²):

Value: 0.56
Interpretation: This value indicates that approximately 56% of the variance in healthy life expectancy can be explained by the model's predictors. An R² value closer to 1 suggests a better fit, while values closer to 0 indicate that the model does not explain much of the variability in the target variable.

## Summary
Overall, the model seems to provide a reasonable fit, explaining a moderate amount of variance in healthy life expectancy. However, the MAE and RMSE indicate that there is still a significant average error in predictions, which suggests room for improvement, possibly through feature engineering, additional data, or trying different modeling techniques.






## Feature Importance

In [10]:
# Get the feature importance array
importances = rf_regressor.feature_importances_

In [11]:
# List the top 10 most important features
imp_features = sorted(zip(importances, X.columns), reverse=True)
top10 = imp_features[:10]
top10

[(0.1317077763485363, 'SDGIHR2021'),
 (0.09730443126288232, 'SDGIPV12M'),
 (0.08171870341360399, 'SDGWSHBOD'),
 (0.0813753226172164, 'MDG_0000000003'),
 (0.07150654899428036, 'UHC_INDEX_REPORTED'),
 (0.07032828197475861, 'MDG_0000000026'),
 (0.06770357031694166, 'HWF_0001'),
 (0.054892263701469074, 'SDGFPALL'),
 (0.04342778624413089, 'FINPROTECTION_CATA_TOT_10_POP'),
 (0.029473154757111302, 'FINPROTECTION_CATA_TOT_25_POP')]

# 2) Same analysis WITH REGION

In [12]:
# Define features set
who_data_region_df = who_data_df.drop(columns=["DIM_GEO_NAME", "DIM_GEO_CODE", "sub-region", "DIM_TIME_YEAR",])
who_data_region_df.head()

region  AMR_INFECT_ECOLI  AMR_INFECT_MRSA  FINPROTECTION_CATA_TOT_10_POP  \
0   Asia               NaN              NaN                            NaN   
1   Asia               NaN              NaN                            NaN   
2   Asia               NaN              NaN                            NaN   
3   Asia               NaN              NaN                            NaN   
4   Asia               NaN              NaN                            NaN   

   FINPROTECTION_CATA_TOT_25_POP  GHED_GGHE-DGGE_SHA2011  GLASSAMC_AWARE  \
0                            NaN                     NaN             NaN   
1                            NaN                     NaN             NaN   
2                            NaN                     NaN             NaN   
3                            NaN                    4.05             NaN   
4                            NaN                     NaN             NaN   

   HWF_0001  HWF_0006  HWF_0010  ...  UHC_INDEX_REPORTED  \
0       NaN       NaN       NaN  ...                 NaN   
1       NaN      4.52       NaN  ...                 NaN   
2       NaN       NaN     0.714  ...                 NaN   
3       NaN       NaN       NaN  ...           40.884609   
4       NaN       NaN       NaN  ...                 NaN   

   VACCINEPREVENTABLE_WILDPOLIO  VIOLENCE_HOMICIDERATE  WHOSIS_000003  \
0                           NaN                    NaN            NaN   
1                           NaN                    NaN            NaN   
2                           NaN                    NaN            NaN   
3                           NaN                8.37823            NaN   
4                           NaN                    NaN      35.489811   

   WHS4_100  WSH_DOMESTIC_WASTE_SAFELY_TREATED  WSH_HYGIENE_BASIC  \
0       NaN                                NaN                NaN   
1       NaN                                NaN                NaN   
2       NaN                                NaN                NaN   
3       NaN                                NaN                NaN   
4      69.0                                NaN          48.214699   

   WSH_SANITATION_SAFELY_MANAGED  WSH_WATER_SAFELY_MANAGED  \
0                            NaN                       NaN   
1                            NaN                       NaN   
2                            NaN                       NaN   
3                            NaN                       NaN   
4                            NaN                   30.0341   

   Healthy life expectancy at birth  
0                         52.924999  
1                         53.575001  
2                         53.900002  
3                         54.549999  
4                         54.875000  

[5 rows x 58 columns]

In [13]:
# Create dummie variables for region
region_dummies = pd.get_dummies(who_data_region_df["region"]).astype(int)
region_dummies.head()

Africa  Americas  Asia  Europe  Oceania
0       0         0     1       0        0
1       0         0     1       0        0
2       0         0     1       0        0
3       0         0     1       0        0
4       0         0     1       0        0

In [14]:
# Join the the region_dummies and the who_data_region_df
who_data_region_df = pd.concat([who_data_region_df, region_dummies], axis=1)

# Drop the region column and the target column to get the the features set
X1 = who_data_region_df.drop(columns=["region", "Healthy life expectancy at birth"])
X1.head()

AMR_INFECT_ECOLI  AMR_INFECT_MRSA  FINPROTECTION_CATA_TOT_10_POP  \
0               NaN              NaN                            NaN   
1               NaN              NaN                            NaN   
2               NaN              NaN                            NaN   
3               NaN              NaN                            NaN   
4               NaN              NaN                            NaN   

   FINPROTECTION_CATA_TOT_25_POP  GHED_GGHE-DGGE_SHA2011  GLASSAMC_AWARE  \
0                            NaN                     NaN             NaN   
1                            NaN                     NaN             NaN   
2                            NaN                     NaN             NaN   
3                            NaN                    4.05             NaN   
4                            NaN                     NaN             NaN   

   HWF_0001  HWF_0006  HWF_0010  HWF_0014  ...  WHS4_100  \
0       NaN       NaN       NaN       NaN  ...       NaN   
1       NaN      4.52       NaN     0.292  ...       NaN   
2       NaN       NaN     0.714       NaN  ...       NaN   
3       NaN       NaN       NaN       NaN  ...       NaN   
4       NaN       NaN       NaN       NaN  ...      69.0   

   WSH_DOMESTIC_WASTE_SAFELY_TREATED  WSH_HYGIENE_BASIC  \
0                                NaN                NaN   
1                                NaN                NaN   
2                                NaN                NaN   
3                                NaN                NaN   
4                                NaN          48.214699   

   WSH_SANITATION_SAFELY_MANAGED  WSH_WATER_SAFELY_MANAGED  Africa  Americas  \
0                            NaN                       NaN       0         0   
1                            NaN                       NaN       0         0   
2                            NaN                       NaN       0         0   
3                            NaN                       NaN       0         0   
4                            NaN                   30.0341       0         0   

   Asia  Europe  Oceania  
0     1       0        0  
1     1       0        0  
2     1       0        0  
3     1       0        0  
4     1       0        0  

[5 rows x 61 columns]

In [15]:
# Define target vector
y1 = who_data_region_df["Healthy life expectancy at birth"]
y1.head()

0    52.924999
1    53.575001
2    53.900002
3    54.549999
4    54.875000
Name: Healthy life expectancy at birth, dtype: float64

In [16]:
# Splitting into Train and Test sets
X1_train, X1_test, y1_train, y1_test = train_test_split(X1, y1, random_state=78)

## Fitting the Random Forest Regressor

In [17]:
# Create the random forest regressor instance
rf_regressor1 = RandomForestRegressor(n_estimators=500, random_state=78)

In [18]:
# Fit the model
rf_regressor1.fit(X1_train, y1_train)

RandomForestRegressor(n_estimators=500, random_state=78)

## Making Predictions Using the Random Forest Regressor Model

In [19]:
# Making predictions using the testing data
predictions1 = rf_regressor1.predict(X1_test)

## Model Evaluation

In [20]:
# Regression evaluation metrics using scikit-learn

# Mean Absolute Error (MAE):
mae1 = mean_absolute_error(y1_test, predictions1)
print(f'Region Mean Absolute Error (MAE): {mae1}')

# Mean Squared Error (MSE):
mse1 = mean_squared_error(y1_test, predictions1)
print(f'Region Mean Squared Error (MSE): {mse1}')

# Root Mean Squared Error (RMSE):
rmse1 = np.sqrt(mean_squared_error(y1_test, predictions1))
print(f'Region Root Mean Squared Error (RMSE): {rmse1}')

# R-squared (R²):
r2a = r2_score(y1_test, predictions1)
print(f'Region R-squared (R²): {r2a}')

Region Mean Absolute Error (MAE): 2.1040831047063255
Region Mean Squared Error (MSE): 8.100681348050198
Region Root Mean Squared Error (RMSE): 2.846169592285428
Region R-squared (R²): 0.7565761746288084


## Feature Importance

In [21]:
# Get the feature importance array
importances1 = rf_regressor1.feature_importances_

In [22]:
# List the top 10 most important features
imp_features1 = sorted(zip(importances1, X1.columns), reverse=True)
top10a = imp_features1[:10]
top10a

[(0.5374283065639109, 'Africa'),
 (0.060233971446588486, 'Europe'),
 (0.03783691652092642, 'SDGIHR2021'),
 (0.02323530947713734, 'MDG_0000000003'),
 (0.02210504497457881, 'MDG_0000000026'),
 (0.021401562331728, 'UHC_INDEX_REPORTED'),
 (0.02050106289354272, 'SDGIPV12M'),
 (0.019665780876539068, 'NCDMORT3070'),
 (0.01961765237145426, 'SDGFPALL'),
 (0.019537783415390808, 'SDGAIRBODA')]

# 3) Same analysis WITH SUB-REGION

In [23]:
# Define features set
who_data_subregion_df = who_data_df.drop(columns=["DIM_GEO_NAME", "DIM_GEO_CODE", "region", "DIM_TIME_YEAR",])
who_data_subregion_df.head()

sub-region  AMR_INFECT_ECOLI  AMR_INFECT_MRSA  \
0  Southern Asia               NaN              NaN   
1  Southern Asia               NaN              NaN   
2  Southern Asia               NaN              NaN   
3  Southern Asia               NaN              NaN   
4  Southern Asia               NaN              NaN   

   FINPROTECTION_CATA_TOT_10_POP  FINPROTECTION_CATA_TOT_25_POP  \
0                            NaN                            NaN   
1                            NaN                            NaN   
2                            NaN                            NaN   
3                            NaN                            NaN   
4                            NaN                            NaN   

   GHED_GGHE-DGGE_SHA2011  GLASSAMC_AWARE  HWF_0001  HWF_0006  HWF_0010  ...  \
0                     NaN             NaN       NaN       NaN       NaN  ...   
1                     NaN             NaN       NaN      4.52       NaN  ...   
2                     NaN             NaN       NaN       NaN     0.714  ...   
3                    4.05             NaN       NaN       NaN       NaN  ...   
4                     NaN             NaN       NaN       NaN       NaN  ...   

   UHC_INDEX_REPORTED  VACCINEPREVENTABLE_WILDPOLIO  VIOLENCE_HOMICIDERATE  \
0                 NaN                           NaN                    NaN   
1                 NaN                           NaN                    NaN   
2                 NaN                           NaN                    NaN   
3           40.884609                           NaN                8.37823   
4                 NaN                           NaN                    NaN   

   WHOSIS_000003  WHS4_100  WSH_DOMESTIC_WASTE_SAFELY_TREATED  \
0            NaN       NaN                                NaN   
1            NaN       NaN                                NaN   
2            NaN       NaN                                NaN   
3            NaN       NaN                                NaN   
4      35.489811      69.0                                NaN   

   WSH_HYGIENE_BASIC  WSH_SANITATION_SAFELY_MANAGED  WSH_WATER_SAFELY_MANAGED  \
0                NaN                            NaN                       NaN   
1                NaN                            NaN                       NaN   
2                NaN                            NaN                       NaN   
3                NaN                            NaN                       NaN   
4          48.214699                            NaN                   30.0341   

   Healthy life expectancy at birth  
0                         52.924999  
1                         53.575001  
2                         53.900002  
3                         54.549999  
4                         54.875000  

[5 rows x 58 columns]

In [24]:
# Create dummie variables for sub-region
subregion_dummies = pd.get_dummies(who_data_subregion_df["sub-region"]).astype(int)
subregion_dummies.head()

Australia and New Zealand  Central Asia  Eastern Asia  Eastern Europe  \
0                          0             0             0               0   
1                          0             0             0               0   
2                          0             0             0               0   
3                          0             0             0               0   
4                          0             0             0               0   

   Latin America and the Caribbean  Northern Africa  Northern America  \
0                                0                0                 0   
1                                0                0                 0   
2                                0                0                 0   
3                                0                0                 0   
4                                0                0                 0   

   Northern Europe  South-eastern Asia  Southern Asia  Southern Europe  \
0                0                   0              1                0   
1                0                   0              1                0   
2                0                   0              1                0   
3                0                   0              1                0   
4                0                   0              1                0   

   Sub-Saharan Africa  Western Asia  Western Europe  
0                   0             0               0  
1                   0             0               0  
2                   0             0               0  
3                   0             0               0  
4                   0             0               0

In [25]:
# Join the the subregion_dummies and the who_data_subregion_df
who_data_subregion_df = pd.concat([who_data_subregion_df, subregion_dummies], axis=1)

# Drop the sub-region column and the target column to get the the features set
X2 = who_data_subregion_df.drop(columns=["sub-region", "Healthy life expectancy at birth"])
X2.head()

AMR_INFECT_ECOLI  AMR_INFECT_MRSA  FINPROTECTION_CATA_TOT_10_POP  \
0               NaN              NaN                            NaN   
1               NaN              NaN                            NaN   
2               NaN              NaN                            NaN   
3               NaN              NaN                            NaN   
4               NaN              NaN                            NaN   

   FINPROTECTION_CATA_TOT_25_POP  GHED_GGHE-DGGE_SHA2011  GLASSAMC_AWARE  \
0                            NaN                     NaN             NaN   
1                            NaN                     NaN             NaN   
2                            NaN                     NaN             NaN   
3                            NaN                    4.05             NaN   
4                            NaN                     NaN             NaN   

   HWF_0001  HWF_0006  HWF_0010  HWF_0014  ...  \
0       NaN       NaN       NaN       NaN  ...   
1       NaN      4.52       NaN     0.292  ...   
2       NaN       NaN     0.714       NaN  ...   
3       NaN       NaN       NaN       NaN  ...   
4       NaN       NaN       NaN       NaN  ...   

   Latin America and the Caribbean  Northern Africa  Northern America  \
0                                0                0                 0   
1                                0                0                 0   
2                                0                0                 0   
3                                0                0                 0   
4                                0                0                 0   

   Northern Europe  South-eastern Asia  Southern Asia  Southern Europe  \
0                0                   0              1                0   
1                0                   0              1                0   
2                0                   0              1                0   
3                0                   0              1                0   
4                0                   0              1                0   

   Sub-Saharan Africa  Western Asia  Western Europe  
0                   0             0               0  
1                   0             0               0  
2                   0             0               0  
3                   0             0               0  
4                   0             0               0  

[5 rows x 70 columns]

In [26]:
# Define target vector
y2 = who_data_subregion_df["Healthy life expectancy at birth"]
y2.head()

0    52.924999
1    53.575001
2    53.900002
3    54.549999
4    54.875000
Name: Healthy life expectancy at birth, dtype: float64

In [27]:
# Splitting into Train and Test sets
X2_train, X2_test, y2_train, y2_test = train_test_split(X2, y2, random_state=78)

## Fitting the Random Forest Regressor

In [28]:
# Create the random forest regressor instance
rf_regressor2 = RandomForestRegressor(n_estimators=500, random_state=78)

In [29]:
# Fit the model
rf_regressor2.fit(X2_train, y2_train)

RandomForestRegressor(n_estimators=500, random_state=78)

## Making Predictions Using the Random Forest Regressor Model

In [30]:
# Making predictions using the testing data
predictions2 = rf_regressor2.predict(X2_test)

## Model Evaluation

In [31]:
# Regression evaluation metrics using scikit-learn

# Mean Absolute Error (MAE):
mae2 = mean_absolute_error(y2_test, predictions2)
print(f'Sub-Region Mean Absolute Error (MAE): {mae2}')

# Mean Squared Error (MSE):
mse2 = mean_squared_error(y2_test, predictions2)
print(f'Sub-Region Mean Squared Error (MSE): {mse2}')

# Root Mean Squared Error (RMSE):
rmse2 = np.sqrt(mean_squared_error(y2_test, predictions2))
print(f'Sub-Region Root Mean Squared Error (RMSE): {rmse2}')

# R-squared (R²):
r2b = r2_score(y2_test, predictions2)
print(f'Sub-Region R-squared (R²): {r2b}')

Sub-Region Mean Absolute Error (MAE): 2.044056805753636
Sub-Region Mean Squared Error (MSE): 7.632701531954053
Sub-Region Root Mean Squared Error (RMSE): 2.762734430225615
Sub-Region R-squared (R²): 0.7706388728311083


## Feature Importance

In [32]:
# Get the feature importance array
importances2 = rf_regressor2.feature_importances_

In [33]:
# List the top 10 most important features
imp_features2 = sorted(zip(importances2, X2.columns), reverse=True)
top10b = imp_features2[:10]
top10b

[(0.599967343087071, 'Sub-Saharan Africa'),
 (0.044774162520667925, 'Southern Asia'),
 (0.029782856846375796, 'Western Europe'),
 (0.023768786279524716, 'SDGIHR2021'),
 (0.01759060680820558, 'Northern Europe'),
 (0.016895579762629113, 'Southern Europe'),
 (0.016383868985779128, 'MDG_0000000003'),
 (0.01416493198673304, 'SDGIPV12M'),
 (0.014045293390930618, 'NCDMORT3070'),
 (0.013654928616886322, 'HWF_0006')]